Compute the correct overflow-x and overflow-y values for table elements

Ahmad-S792 opened the pull request:
#### e6264aebbca8560834e3cf399543ab4dbbbc4855
<pre>
Compute the correct overflow-x and overflow-y values for table elements

<a href="https://bugs.webkit.org/show_bug.cgi?id=252422">https://bugs.webkit.org/show_bug.cgi?id=252422</a>
rdar://problem/105850323

Reviewed by Antti Koivisto.

This patch is to align WebKit with Chromium / Blink and Firefox / Gecko.

Inspired &amp; Test Merge: <a href="https://src.chromium.org/viewvc/blink?view=revision&amp">https://src.chromium.org/viewvc/blink?view=revision&amp</a>;revision=199640

This patch is to align WebKit to respect web-specification[1] &amp; [2], which tells us that
display:table must treat anything other than hidden as visible and if overflow-x and
overflow-y are different, how they must be resolved. This latter part is already taken care
of, we&apos;re just allowing it to be applied to tables and table parts now.
Additionally, it allows us to have consolidated logic and simplify our code.

[1] <a href="https://drafts.csswg.org/css2/#overflow">https://drafts.csswg.org/css2/#overflow</a>
[2] <a href="https://drafts.csswg.org/css-overflow-3/#propdef-overflow-x">https://drafts.csswg.org/css-overflow-3/#propdef-overflow-x</a>

* Source/WebCore/style/StyleAdjuster.cpp:
(isOverflowClipOrVisible): New static &apos;bool&apos; helper function
(Adjuster::adjust): Update handling of &quot;overflow-x&quot; and &quot;overflow-y&quot; align with spec and
interop with other engines and simplify code with comments
* LayoutTests/fast/table/table-different-overflow-values.html: Add Test Case
* LayoutTests/fast/table/table-different-overflow-values-expected.txt: Add Test Case Expectation
* LayoutTests/fast/table/table-different-overflow-values-2.html: Add Test Case
* LayoutTests/fast/table/table-different-overflow-values-2-expected.txt: Add Test Case Expectation
* LayoutTests/fast/table/overflowScroll.html: Add Test Case
* LayoutTests/fast/table/overflowScroll-expected.html: Add Test Case Expectation
* LayoutTests/fast/table/overflowScroll-display-block.html: Add Test Case
* LayoutTests/fast/table/overflowScroll-display-block-expected.html: Add Test Case Expectation
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/overflow/computed-value-001-expected.txt: Rebaselined

Canonical link: <a href="https://commits.webkit.org/266560@main">https://commits.webkit.org/266560@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/47e4dbbc2798d897a9be5642aad15c430dc0600a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/14168 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/14479 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/14816 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/15903 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/13425 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/14269 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/16990 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/14560 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/16097 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/14342 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/14910 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/12010 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/16622 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/39/builds/12193 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/12771 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/19791 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/13272 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/12935 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/16140 "Passed tests") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/7/builds/13484 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/11340 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/12763 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/12631 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/4/builds/17098 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/1680 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/31/builds/13327 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

anttijk commented on LayoutTests/fast/table/overflowScroll-display-block-expected.html:
```diff
@@ -0,0 +1,17 @@
+</div>
+<div style="overflow: scroll;">
+</div>
+<p> crbug.com/368699: When table elements have display: block, they should be allowed to have overflow:scroll. The exception is the table element, which doesn't permit anything other than visible per http://dev.w3.org/csswg/css2/visufx.html#overflow.</p>
```
It would be nice to make WPT versions of these old Chrome tests.

Ahmad-S792 replied:
Good idea! Let me try it later today. If not, I will push this one and deal with tests directly on WPT and then add them here.

Ahmad-S792 replied:
Part 1 to sync easy one to WPT - PR raised. Will do others later. Closing this right now and merging.

Ahmad-S792 committed:
Compute the correct overflow-x and overflow-y values for table elements

https://bugs.webkit.org/show_bug.cgi?id=252422
rdar://problem/105850323

Reviewed by Antti Koivisto.

This patch is to align WebKit with Chromium / Blink and Firefox / Gecko.

Inspired & Test Merge: https://src.chromium.org/viewvc/blink?view=revision&revision=199640

This patch is to align WebKit to respect web-specification[1] & [2], which tells us that
display:table must treat anything other than hidden as visible and if overflow-x and
overflow-y are different, how they must be resolved. This latter part is already taken care
of, we're just allowing it to be applied to tables and table parts now.
Additionally, it allows us to have consolidated logic and simplify our code.

[1] https://drafts.csswg.org/css2/#overflow
[2] https://drafts.csswg.org/css-overflow-3/#propdef-overflow-x

* Source/WebCore/style/StyleAdjuster.cpp:
(isOverflowClipOrVisible): New static 'bool' helper function
(Adjuster::adjust): Update handling of "overflow-x" and "overflow-y" align with spec and
interop with other engines and simplify code with comments
* LayoutTests/fast/table/table-different-overflow-values.html: Add Test Case
* LayoutTests/fast/table/table-different-overflow-values-expected.txt: Add Test Case Expectation
* LayoutTests/fast/table/table-different-overflow-values-2.html: Add Test Case
* LayoutTests/fast/table/table-different-overflow-values-2-expected.txt: Add Test Case Expectation
* LayoutTests/fast/table/overflowScroll.html: Add Test Case
* LayoutTests/fast/table/overflowScroll-expected.html: Add Test Case Expectation
* LayoutTests/fast/table/overflowScroll-display-block.html: Add Test Case
* LayoutTests/fast/table/overflowScroll-display-block-expected.html: Add Test Case Expectation
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/overflow/computed-value-001-expected.txt: Rebaselined

Canonical link: https://commits.webkit.org/266560@main
